File: jira_flutter_java/lib/Features/Project/ProjectModel/project_response.dart
class ProjectResponse {
  final int? id;
  final String name;
  final String? description;
  final DateTime? deadline;
  final String? creatorUid;
  final List<String>? memberUids;
  final DateTime? createdAt;

  ProjectResponse({
    this.id,
    required this.name,
    this.description,
    this.deadline,
    this.creatorUid,
    this.memberUids,
    this.createdAt,
  });

  factory ProjectResponse.fromJson(Map<String, dynamic> json) {
    return ProjectResponse(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      deadline: json['deadline'] != null
          ? DateTime.parse(json['deadline'])
          : null,
      creatorUid: json['creatorUid'],
      memberUids: (json['memberUids'] as List?)
          ?.map((e) => e.toString())
          .toList(),
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
    );
  }
}

File: backend/jira-backend/src/main/java/com/jira/backend/dto/ProjectResponse.java
package com.jira.backend.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Builder
public class ProjectResponse {
    private Long id;
    private String name;
    private String description;
    private LocalDateTime deadline;
    private String creatorUid;
    private List<String> memberUids;
    private LocalDateTime createdAt;
}
